lldb/source/Plugins/LanguageRuntime/Swift/ReflectionContext.cpp

```diff
@@ -57,8 +57,24 @@ struct DescriptorFinderForwarder : public swift::reflection::DescriptorFinder {
 
   void ClearExternalDescriptorFinder() { m_descriptor_finder = nullptr; }
 
+  void SetImageAdded(bool image_added) {
+    m_image_added |= image_added;
+  }
+
 private:
+
+  bool shouldConsultDescriptorFinder() {
+    switch (Target::GetGlobalProperties().GetSwiftEnableFullDwarfDebugging()) {
+      case lldb_private::eAutoBoolTrue:
+        return true;
+      case lldb_private::eAutoBoolFalse:
+        return false;
+      case lldb_private::eAutoBoolAuto:
+        return !m_image_added;
```

Comment:
Can you comment how this works here? It's not obvious why `m_image_added` is the condition here.